src/main/java/com/o19s/es/ltr/feature/store/PrecompiledExpressionFeature.java

```diff
@@ -86,8 +125,8 @@ public int hashCode() {
 
     @Override
     public void validate(FeatureSet set) {
-        for(String var : expression.variables) {
-            if(!set.hasFeature(var)) {
+        for (String var : expression.variables) {
```

Comment:
There should be something to make sure that we do not override something here.
E.g.
- you feature set has feature `foo`
- you add a parameter named `foo`

We should either fails or explicitly mention somewhere that there is some precedence (e.g. query param always wins).

Reply:
Chose to fail in case of overlap in param name and feature name.